WookieeMatt/Essence20
# Upgrade Drop

Created ability to drop Upgrades on Armor and Weapons
Added Ability to delete upgrades from sheets. 
Right now you can only drop an upgrade on once that might need to change. 

Testing Steps
Create Armor Upgrade make sure it can drop on Armor Items but not Weapon Items, and that it can be deleted. 
Create Weapon Upgrade make sure it can drop on Weapon Items but not Armor Items, and that it can be deleted. 
Create Drone Upgrade and validate it can't drop on either.
